modules/waypoint_tracking.py

```diff
@@ -3,18 +3,19 @@
 including the current waypoint sequence, location, and destination.
 """
 
-import dronekit
+# import dronekit
+from modules.common.mavlink.modules.flight_controller import FlightController
 
 
 def get_current_waypoint_info(
-    drone: dronekit.Vehicle,
+    drone: FlightController,
 ) -> "tuple[bool, tuple[int, tuple[float, float] | None] | None]":
     """
     Function to retrieve information about the current waypoint sequence and destination
 
     Parameters
     ----------
-    drone: dronekit.Vehicle
+    drone: FlightController
         The connected drone.
 
     Returns
```

Comment:
drone.commands.download() and drone.commands.wait_ready() would not run right now (remove them).

since drone is a flightcontroller instead of a dronekit_vehicle. you would have to do drone.drone to access the dronekit vehicle. but regardless we do not want to access the internal dronekit instance of the flightcontroller. the lines